### ISSUE TITLE
Implement `enum` formatting in Ocean's Formatter

### ISSUE THREAD

**Issue by Geod24** [bosagora/agora#725]
Since Ocean dropped D1 compatibility, it's now possible to do this, and would simplify our live a bit.

*Definition of Done:*
Basically:
```D
enum Foo
{
    A,
    B,
    C,
    D,
}

import ocean.text.convert.Formatter;
unittest
{
    assert(format("{} {} {} {}", Foo.A, Foo.B, Foo.C, Foo.D) == "Foo.A Foo.B Foo.C Foo.D");
    Foo invalid1 = cast(Foo) 42;
    assert(format("{}", invalid1) == "cast(Foo) 42");
}
```

**Comment by MichaelKim20** [comment]
@Geod24 Can I use `std.conv`?

**Comment by Geod24** [comment]
No, Ocean cannot depend on Phobos

**Comment by MichaelKim20** [comment]
What would I use as an alternative to 'std.conv'?
I've tried ocean.util.Convert, and there's a compilation error.

**Comment by Geod24** [comment]
The goal of this issue is to implement, in the `Formatter`, the same kind of conversion that is done in `std.conv`, but without using it, and without needlessly allocating memory.
This should be done using compile time introspection features of the language only (no `std.traits`). There's multiple approaches to it, so you have to find which one is most efficient. You can get the names of each members of the enum with `__traits(allMembers, NameOfEnumType)`, and play around with `mixin`s.
What I expect it'll look like is a compile-time-generated `switch` / `case`, the `case` being generated at compile time, that can be used with a runtime value.

**Comment by MichaelKim20** [comment]
PR is https://github.com/bpfkorea/ocean/pull/3

**Comment by Geod24** [comment]
Fixed in #752
